File: 剑指offer/62. 圆圈中最后剩下的数字.py
#############约瑟夫环问题
class Solution:
    def lastRemaining(self, n: int, m: int) -> int:
        final_index = 0
        for len_num in range(2, n+1): # i是数组长度
            final_index = (final_index+m)%len_num
        return final_index
        
https://leetcode-cn.com/problems/yuan-quan-zhong-zui-hou-sheng-xia-de-shu-zi-lcof/solution/javajie-jue-yue-se-fu-huan-wen-ti-gao-su-ni-wei-sh/
https://leetcode-cn.com/problems/yuan-quan-zhong-zui-hou-sheng-xia-de-shu-zi-lcof/solution/yuan-quan-zhong-zui-hou-sheng-xia-de-shu-zi-by-lee/

File: 剑指offer/C++_Version/62. 圆圈中最后剩下的数字.cpp
//方法1
class Solution {
public:
    int lastRemaining(int n, int m) 
    {
        int f = 0;
        for(int i=2; i<n+1; i++)
        {
            f = (f + m) % i;
        }
        return f;
    }
};


//方法2
class Solution {
public:
    int lastRemaining(int n, int m) 
    {
        int final_index = 0;
        for(int len_num=2; len_num<n+1; len_num++)
        {
            final_index = (final_index+m) % len_num;
        }
        return final_index;
    }
};
